hw/ip/i2c/doc/theory_of_operation.md

```diff
@@ -203,69 +301,63 @@ These timing parameters are then fed directly to the I2C state machine to contro
+For these reasons, [`TIMEOUT_CTRL`](registers.md#timeout_ctrl) provides a timeout mechanism for determining one of two variants of an SCL low timeout.
 
-#### Stretching during read
-- When a target device receives a start and read command, it may stretch the clock for either of the following two reasons.
-  - If there is no data available to be sent back (TX FIFO empty case), the target stretches the clock until data is made available by software.
-  - If there is more than 1 entry in the ACQ FIFO.
-    - Having more than 1 entry in the ACQ FIFO suggests there is potentially an unhandled condition (STOP / RESTART) or an unhandled command (START) that requires software intervention before the read can proceed.
-- The `tx_stretch` interrupt is generated to alert software to such a situation.
+Enabling a timeout at all requires setting [`TIMEOUT_CTRL.EN`](registers.md#timeout_ctrl) and configuring [`TIMEOUT_CTRL.VAL`](registers.md#timeout_ctrl) with the desired number of cycles to count before timing out.
+With the configuration of [`TIMEOUT_CTRL.MODE`](registers.md#timeout_ctrl), one of two types of timeouts can be selected.
 
-### Target ACK Control Mode
+If a `STRETCH_TIMEOUT` is selected, the Controller Module detects when some other device holds SCL low for more cycles than than [`TIMEOUT_CTRL.VAL`](registers.md#timeout_ctrl) clock ticks.
+If the counter expires, the `stretch_timeout` interrupt will be asserted.
 
-ACK Control Mode is a feature that allows software to take control of ACK/NACK decisions during a Write transfer.
-When ACK Control Mode is disabled in [`CTRL.ACK_CTRL_EN`](registers.md#ctrl), the Target module will automatically ACK any incoming data byte, so long as there is sufficient space in the ACQ FIFO.
-If there isn't enough space for the data byte and a STOP, the Target module will stretch the clock until space is made available (or a stretch timeout occurs).
-However, if ACK Control Mode is enabled, the [`TARGET_ACK_CTRL`](registers.md#target_ack_ctrl) CSR can also cause the Target module to stretch the clock at the (N)ACK phase.
+If a `BUS_TIMEOUT` is selected, the SCL low time accumulates across all sources, including this IP's Controller Module.
+This counter mode can be used to detect SMBus's definition of a bus timeout.
+If a bus timeout occurs while the Controller Module is active, the corresponding [`CONTROLLER_EVENTS.BUS_TIMEOUT`](registers.md#controller_events) bit is set, halting the Controller Module and requiring handling from software before the next command will process.
+<!-- TODO: Bus timeouts to the target aren't qualified by whether the particular transaction addressed our target, leading to ACQ FIFO usage. Getting some kind of notification that a bus timeout occurred might be good, but consider whether all responses are appropriate when the transaction doesn't address us. The triggering conditions are awkward. -->
```

Comment:
Bus timeout handling seems to be a bit awkward. We may want to create an issue about making it better, if possible--We'd have to decide what actually would be the ideal response, though!